tests/service/Symbols.fs

```diff
@@ -170,6 +170,68 @@ let x = 123
         |> Option.map (fun su -> su.Symbol :?> FSharpMemberOrFunctionOrValue)
         |> Option.iter (fun symbol -> symbol.Attributes.Count |> shouldEqual 1)
 
+// FS-1112
+module ObsoleteAttributeTest =
```

Comment:
Not sure where tests should be (probably in `fsharpqa`?). This is their temporary location

Reply:
ComponentTests suite sounds fitting. I would avoid adding new tests in fsharpqa, if not necessary, it's hard to debug, and run only on desktop framework.

Reply:
@vzarytovskii , thanks, I definitely need help! 😅 

To implement the feature, I *think* I need to extract the info about the caller from `env : TcEnv` and exclude generated `obsolete`-related warnings. But I only could find `eCallerMemberName` which has the *string* name of the caller. What I want is some... syntax entity from a tree?

I think after getting the caller I need to see if the caller, or its *parent* or enclosing entity, or etc. is `obsolete`.

So could you tell me where I could learn more about this tree analysis from the `env` variable? Or maybe I should take some other approach?

I admit I might be asking silly questions, but I'm new to the compiler. I briefly read the [compiler guide](https://github.com/dotnet/fsharp/blob/main/docs/compiler-guide.md), but not sure I found all the answers 😅 (although some points became clearer).

Reply:
@vzarytovskii hey? 😅